internal/envoy/v3/listener.go

```diff
@@ -568,6 +568,12 @@ func (b *httpConnectionManagerBuilder) Get() *envoy_config_listener_v3.Filter {
 				UpgradeType: "websocket",
 			},
 		)
+		http2Options.AllowConnect = true
+	}
+
+	// Assign http2Options only if it has been modified
+	if b.http2MaxConcurrentStreams != nil {
```

Comment:
i dont think this is quite right? if max concurrent streams is not set, but websockets are enabled we would still want to set the http2 options